src/Build/BackEnd/Components/SdkResolution/NuGet/NuGetSdkResolver.cs

```diff
@@ -0,0 +1,202 @@
+        /// This method should not consume any NuGet classes directly to avoid loading additional assemblies when they are not needed.  This method
+        /// returns an object so that NuGetVersion is not consumed directly.
+        /// </summary>
+        internal static bool TryGetNuGetVersionForSdk(string id, string version, SdkResolverContextBase context, out object parsedVersion)
+        {
+            if (!String.IsNullOrWhiteSpace(version))
+            {
+                // Use the version specified in the project if it is a NuGet compatible version
+                return NuGetAbstraction.TryParseNuGetVersion(version, out parsedVersion);
+            }
+
+            Dictionary<string, string> msbuildSdkVersions;
+
+            // Get the SDK versions from a previous state, otherwise find and load global.json to get them
+
+            if (context.State is Dictionary<string, string>)
+            {
+                msbuildSdkVersions = (Dictionary<string, string>) context.State;
+            }
+            else
+            {
+                msbuildSdkVersions = GlobalJsonReader.GetMSBuildSdkVersions(context);
+            }
+
+            // Save the SDK versions in case this resolver is called again for another SDK in the same build
+            context.State = msbuildSdkVersions;
+
+            string globalJsonVersion;
+
+            // Check if global.json specified a version for this SDK and make sure its a version compatible with NuGet
+            if (msbuildSdkVersions != null && msbuildSdkVersions.TryGetValue(id, out globalJsonVersion) && !String.IsNullOrWhiteSpace(globalJsonVersion))
+            {
+                return NuGetAbstraction.TryParseNuGetVersion(globalJsonVersion, out parsedVersion);
+            }
+
+            parsedVersion = null;
+
+            return false;
+        }
+
+        /// <summary>
+        /// IMPORTANT: This class is used to ensure that <see cref="NuGetSdkResolver"/> does not consume any NuGet classes directly.  This ensures that no NuGet assemblies
+        /// are loaded unless they are needed.  Do not implement anything in <see cref="NuGetSdkResolver"/> that uses a NuGet class and instead place it here.
+        /// </summary>
+        private static class NuGetAbstraction
+        {
+            public static SdkResultBase GetSdkResult(SdkReference sdk, object nuGetVersion, SdkResolverContextBase context, SdkResultFactoryBase factory)
+            {
+                // Cast the NuGet version since the caller does not want to consume NuGet classes directly
+                NuGetVersion parsedSdkVersion = (NuGetVersion) nuGetVersion;
+
+                // Stores errors and warnings for the result
+                ICollection<string> errors = new List<string>();
+                ICollection<string> warnings = new List<string>();
+
+                // Load NuGet settings and a path resolver
+                ISettings settings = Settings.LoadDefaultSettings(context.ProjectFilePath);
```

Comment:
Under what circumstances would this need _non_-default settings?

Reply:
The "default" settings means to find the settings in the default way.  Reference: https://docs.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#how-settings-are-applied

At this time, there would be no way for a user to override how settings are loaded.